### PR TITLE
Create a service for products

### LINKED ISSUE #7: Create Product details subcomponent
Product details should be displayed when the user 'clicks' on one product.

When viewing a product in the App the user could see the state of this product for the supermarkets in the vicinity. 

The `Product Details` component should thus display the title and description (possibly cathegories) of the product and a map view of local supermarkets where this product could be bought.

### LINKED ISSUE #15: create a service for products
Create a small service that communicates our product-related data with the api.

### LINKED ISSUE #48: Create Crud-operations service for the api
Create a basic crud service for the api, that implements the methods `create`,`read`,`update`,`delete`.

### LINKED ISSUE #51: Update Product.vue
Update Product.vue component in a way that it contains the Details component and renders it, depending on the Store value of a `selected` Product.

### LINKED ISSUE #52: Fix Styling for ProductView.
Create Styles for Products View, in a way that viewing the page on mobile dive renders 2 columns where a card expands over 100% of the screen width, tablet (and larger screen) will render 4 columns where a card on click expands to 50-60% screen width.

### LINKED ISSUE #53: create vuex-store
create a `vuex-store` to make state-management possible in the application

### LINKED ISSUE #54: create product store
Create a `vuex-simple` store for our products. It should store data that the application needs for our products (e.g. selected Product, all available products etc.).

### PR DESCRIPTION
**Pull request by spmse**
Adds a Service and corresponding `vuex` Store for Products. Closes #15 
Please wait before merge until WIP is resolved.

### CODE REVIEW

**Comment by spmse** [comment]
WIP will be resolved when #51 can be closed.

**Comment by spmse** [comment]
commit 2f6a402 fixes #51

File: src/services/api/productService.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { BaseApiService } from "./base";
+import { AxiosResponse } from 'axios';
+import { IProduct } from '@/interfaces';
+export class ProductService {
+    
+    static getAllProducts = async (): Promise<AxiosResponse> => {
+        return await BaseApiService.instance.read("/products");
```
Comment (tlahmann) [inline src/services/api/productService.ts:7]: Either embed this line in a try catch block or split this line into two. [ES-lint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) & [another example](https://jakearchibald.com/2017/await-vs-return-vs-return-await/#return-awaiting).

The same applies to all return await calls.

File: src/services/api/base.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  static get instance() {
+    if (!this._instance) {
+      this._instance = new BaseApiService();
+    }
+    return this._instance;
+  }
+
+  read = async (resource: string = ""): Promise<AxiosResponse> => {
+    return await this.http.get(resource);
```
Comment (tlahmann) [inline src/services/api/base.ts:31]: Either embed this line in a try catch block or split this line into two. [ES-lint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) & [another example](https://jakearchibald.com/2017/await-vs-return-vs-return-await/#return-awaiting).

The same applies to all return await calls.

**Review by tlahmann — CHANGES_REQUESTED** [review]

**Comment by tlahmann** [comment]
@SvenPatrickMeier, Is this ready to merge?

**Comment by spmse** [comment]
@tlahmann Almost ready. Will finish tomorrow!

**Comment by spmse** [comment]
Ready to merge after review.

**Review by tlahmann — APPROVED** [review]

**Pull request merged**